system/adb/adb_main.c

```diff
@@ -49,7 +49,18 @@ void adb_log_impl(FAR const char *func, int line, FAR const char *fmt, ...)
 void adb_reboot_impl(const char *target)
 {
 #ifdef CONFIG_BOARDCTL_RESET
-  boardctl(BOARDIOC_RESET, 0);
+  if (strcmp(target, "recovery") == 0)
+    {
+      boardctl(BOARDIOC_RESET, CONFIG_ADBD_RESET_RECOVERY);
+    }
+  else if (strcmp(target, "bootloader") == 0)
+    {
+      boardctl(BOARDIOC_RESET, CONFIG_ADBD_RESET_BOOTLOADER);
+    }
+  else
+    {
+      boardctl(BOARDIOC_RESET, 0);
```

Comment:
Should we `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` or keep `0`?

Reply:
I just see calls `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` all over the apps code. Maybe we should modify it to `boardctl(BOARDIOC_RESET, 0);` in other places in next PR? What do you think?

Reply:
I just see calls `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` all over the apps code. Maybe we should modify it to `boardctl(BOARDIOC_RESET, 0);` in other places in next PR? What do you think?

Reply:
Done.